nvflare/apis/fl_exception.py

```diff
@@ -16,12 +16,14 @@
 class FLCommunicationError(Exception):
     """Base class for fed_learn communication exceptions."""
 
-    def __init__(self, exception, message):
+    def __init__(self, message, exception=None):
         """Init the FLCommunicationError.
 
         Args:
             exception: grpc.RpcError when trying to register gprc channel
         """
         super().__init__()
-        # Copy all the gRPC exception properties into FLCommunicationError instance.
-        self.__dict__.update(exception.__dict__)
+        # Copy all the exception properties into FLCommunicationError instance.
+        if exception:
+            self.__dict__.update(exception.__dict__)
+        self.message = message
```

Comment:
Where are we using this?

If we modify the "__init__" method here we need to also change all the places that use this class.

Reply:
This has already been changed.